### DISCUSSION TITLE
blocksByName

### DISCUSSION THREAD

**Discussion by AshDoesStuf** [PrismarineJS/mineflayer#2635] [Q&A]
when i run the bot it give me this error
TypeError: Cannot read properties of null (reading 'blocksByName')

how to fix?

**Comment by amoraschi** [comment]
Show code, should be from minecraft-data

**Reply by AshDoesStuf** [reply]
function gotoChest() {
  const mcData = require('minecraft-data')(bot.version)
  const movements = new Movements(bot, mcData)
  movements.scafoldingBlocks = []
  bot.pathfinder.setMovements(movements)

  const chest = bot.findBlock({
    matching: mcData.blocksByName.chest.id,
    maxDistance: 16
  })

  if (!chest) {
    bot.chat('no chest found')
    return
  }

  const x = chest.position.x +1
  const y = chest.position.y
  const z = chest.position.z
  const goal = new GoalBlock(x, y, z) 
  bot.pathfinder.setGoal(goal)

  if (chest) {
    try {
      bot.openChest(chest)
      chest.deposit({
        items: bot.inventory.items()
      })
    }

    catch (err) {
      bot.chat('error opening chest')
    }
  }
  chest.close()

}

**Comment by amoraschi** [comment]
Error should be self explanatory, `mcData` is null, maybe it failed to load, you should load it after the bot has spawned
From where did you get that code? That's not how `chest.deposit` works, and `openChest` is deprecated`

You can also join the Discord server for faster replies

**Reply by NoobKeksTV** [reply]
im having same issues, 
tried loading the addons after it has spawned still gave me those issues. even with a delay it doesnt seem to work.

**Reply by A4P5** [reply]
late reply, but did you find fix?
